src/solidity-frontend/solidity_convert.cpp

```diff
@@ -5087,6 +5090,54 @@ bool solidity_convertert::get_expr(
     side_effect_expr_function_callt call;
     const nlohmann::json &callee_expr_json = expr["expression"];
 
+    // * check if it's a recursive function call
+    if (
+      callee_expr_json.contains("referencedDeclaration") &&
+      callee_expr_json["referencedDeclaration"].is_number())
+    {
+      int referenced_decl =
+        callee_expr_json["referencedDeclaration"].get<int>();
+
+      int current_func_id = -1;
+      if (current_functionDecl && (*current_functionDecl).contains("id"))
+      {
+        current_func_id = (*current_functionDecl)["id"].get<int>();
+      }
+
+      if (referenced_decl == current_func_id)
+      {
+        std::string func_id;
+        if (current_functionDecl)
+        {
+          get_function_definition_name(
+            *current_functionDecl, current_functionName, func_id);
+        }
+
+        if (func_id.empty() || context.find_symbol(func_id) == nullptr)
+        {
+          log_error("Cannot find current function symbol");
+          return true;
+        }
+
+        call.function() = symbol_expr(*context.find_symbol(func_id));
```

Comment:
Okay, I don't get it. Why do you need these codes here? It really seems to be a normal function call to me.